worker/server_state.go

```diff
@@ -77,25 +77,27 @@ func setBadgerOptions(opt badger.Options, wal bool) badger.Options {
 	// saved by disabling it.
 	opt.DetectConflicts = false
 
-	glog.Infof("Setting Badger Compression Level: %d", Config.BadgerCompressionLevel)
-	// Default value of badgerCompressionLevel is 3 so compression will always
-	// be enabled, unless it is explicitly disabled by setting the value to 0.
-	if Config.BadgerCompressionLevel != 0 {
-		// By default, compression is disabled in badger.
-		opt.Compression = options.ZSTD
-		opt.ZSTDCompressionLevel = Config.BadgerCompressionLevel
-	}
-
 	var badgerTables string
 	var badgerVlog string
 	if wal {
 		// Settings for the write-ahead log.
 		badgerTables = Config.BadgerWalTables
 		badgerVlog = Config.BadgerWalVlog
+		// Disable compression for WAL as it is supposed to be fast. Compression makes it a
+		// little slow (Though we save some disk space but it is not worth the slowness).
+		opt.Compression = options.None
 	} else {
 		// Settings for the data directory.
 		badgerTables = Config.BadgerTables
 		badgerVlog = Config.BadgerVlog
+		glog.Infof("Setting Badger Compression Level: %d", Config.BadgerCompressionLevel)
```

Comment:
this is more than what the PR description. Do we need this in this PR.

Reply:
This is just a refactor. As we have `if-else` block for pstore/wstore, so having the compression settings set within that block makes more sense I think.
So which earlier used to be a common code for p/w, now is separate for both.

Reply:
No. I meant that earlier there was no logic but now you are looking at the CompressionLevel and setting some values in badger options.